packages/eslint-plugin/tests/rules/explicit-function-return-type.test.ts

```diff
@@ -180,6 +180,46 @@ class App {
       `,
       options: [{ allowTypedFunctionExpressions: true }],
     },
+    // https://github.com/typescript-eslint/typescript-eslint/issues/7552
+    {
+      code: `
+const Comp: FC = () => {
+  return <button onClick={() => {}} />;
+};
+      `,
+      options: [{ allowTypedFunctionExpressions: true }],
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
+    {
+      code: `
+const Comp: FC = () => {
+  return <Button on={{ click: () => {} }} />;
+};
+      `,
+      options: [{ allowTypedFunctionExpressions: true }],
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
+    {
+      code: `
+const Comp: FC = () => {
+  return <button {...{ onClick: () => {} }} />;
+};
+      `,
+      options: [{ allowTypedFunctionExpressions: true }],
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
```

Comment:
for completeness can we also get this as an `invalid` test?

Reply:
Added negated counterparts of all new tests